### PR TITLE
[DO NOT MERGE] Whole-pot calculator (redux)

### PR DESCRIPTION
**Pull request by stevenwilkin**
This is the new model to drive the calculator app. The values [in the spreadsheet](https://docs.google.com/spreadsheets/d/1qAXyJ-a97KdiBz5pxbhE-4_I5FkVljj21CTBGquzheI/edit#gid=0) are used as acceptance criteria.

Any feedback on this would be great!

### CODE REVIEW

File: app/assets/javascripts/whole_pot_calculator.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
+(function() {
+  'use strict';
+
+  function WholePotCalculator(income, pot) {
```
Comment (andrewgarner) [inline app/assets/javascripts/whole_pot_calculator.js]: Maybe `TakeWholePotCalculator` as the slug for the equivalent content is `take-whole-pot`?

Comment (stevenwilkin) [inline app/assets/javascripts/whole_pot_calculator.js]: Sure, that follows on logically

File: app/assets/javascripts/whole_pot_calculator.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    this.taxablePotPortion = 0.75;
+  }
+
+  WholePotCalculator.prototype.round = function(num) {
+    return Math.round(num * 100) / 100;
+  };
+
+  WholePotCalculator.prototype.personalAllowanceFor = function(totalIncome) {
+    if(totalIncome <= 100000) {
```
Comment (andrewgarner) [inline app/assets/javascripts/whole_pot_calculator.js]: `100000` is a magic number. A constant that revealed what this number means such as `ADJUSTED_NET_INCOME_LIMIT` would be good.

File: app/assets/javascripts/whole_pot_calculator.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      return this.standardPersonalAllowance;
+    } else {
+      var reduction = Math.min(Math.floor((totalIncome - 100000) / 2), this.standardPersonalAllowance);
```
Comment (andrewgarner) [inline app/assets/javascripts/whole_pot_calculator.js]: Why do we divide by 2?

File: app/assets/javascripts/whole_pot_calculator.js [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+      return this.standardPersonalAllowance - reduction;
+    }
+  };
+
+  WholePotCalculator.prototype.marginalTaxForIncomeWithAllowance = function(income, allowance) {
```
Comment (andrewgarner) [inline app/assets/javascripts/whole_pot_calculator.js]: Does this belong under the `WholePotCalculator` namespace? These are functions for calculating income tax regardless of the source of income.

File: app/assets/javascripts/whole_pot_calculator.js (hunk 3 above)
Comment (stevenwilkin) [inline app/assets/javascripts/whole_pot_calculator.js]: https://www.gov.uk/income-tax-rates/income-over-100000

File: app/assets/javascripts/whole_pot_calculator.js (hunk 4 above)
Comment (stevenwilkin) [inline app/assets/javascripts/whole_pot_calculator.js]: Strictly it probably shouldn't be part of the public API but it was the simplest thing I could focus on at the start of development

File: app/assets/javascripts/whole_pot_calculator.js (hunk 3 above)
Comment (andrewgarner) [inline app/assets/javascripts/whole_pot_calculator.js]: Assigning `2` to a variable would make this clearer. `REDUCTION_RATIO`?

**Comment by stevenwilkin** [comment]
Updated to remove use of magic numbers when determining personal allowance and have resolved `jscs` errors.

**Comment by stevenwilkin** [comment]
Updated to rename main module to `TakeWholePotCalculator`

**Comment by stevenwilkin** [comment]
Updated to calculate the post-tax/net pot size

**Comment by stevenwilkin** [comment]
Renamed this branch to [whole-pot-calculator-javascript-model)](https://github.com/guidance-guarantee-programme/pension_guidance/tree/whole-pot-calculator-javascript-model) for future reference

**Pull request closed without merging**